pkg/vsphere/extraconfig/keys.go

```diff
@@ -262,3 +262,114 @@ func addPrefixToKey(header, prefix, key string) string {
 	// guestinfoPrefix is const so adding it to the format string directly
 	return fmt.Sprintf(header+"%c%s%s", separator, modifiedPrefix, base)
 }
+
+func calculateKeys(v reflect.Value, field string, prefix string) []string {
+	log.Debugf("v=%#v, field=%#v, prefix=%#v", v, field, prefix)
+	if v.Kind() == reflect.Ptr {
+		return calculateKeys(v.Elem(), field, prefix)
+	}
+
+	if field == "" {
+		return []string{prefix}
+	}
+
+	s := strings.SplitN(field, ".", 2)
```

Comment:
I'm new to this guest info stuff.  Is this a field separator inherent to extraconfig?  If so does this conflict or break with this (https://github.com/vmware/vic/pull/3062/files)?